tracer/src/Datadog.Trace/Telemetry/Collectors/MetricsTelemetryCollectorBase.cs

```diff
@@ -223,12 +223,12 @@ private async Task AggregateMetricsLoopAsync()
             return _wafAndRulesVersionTags ?? _unknownWafAndRulesVersionTags;
         }
 
-        if (string.Equals(metricKeyTags[0], "waf_version"))
+        if (metricKeyTags[0].StartsWith("waf_version"))
```

Comment:
I'm confused, I'm 90% sure this shouldn't be required 😕 `metricKeyTags` contains the values from the metric buffers, e.g.

https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Generated/net461/Datadog.Trace.SourceGenerators/TelemetryMetricGenerator/MetricsTelemetryCollector_Count.g.cs#L575-L577

which should be _constant_

Reply:
Discussed offline, it's because we're replacing the "original" value in the array. 

Previously, when we assumed that _all_ ASM metrics had `waf_Version`, we just replaced it without checking. After our recent fix, we added a check, which is why we started hitting this.

So using `StartsWith()` isn't that clean, but it's still the best option we have right now. Cleaner alternatives would be for _all_ ASM metrics to have these two tags (for example) which we can potentially address later if needs be